src/language/nodes.py

```diff
@@ -242,63 +244,63 @@ def _is_member_type_wrapped_as_shared_pointer(self):
     def member_rvalue_typename(self):
         """returns rvalue reference type when used as returned or parameter type"""
         typename = self.member_typename
-        if not self.is_integral_type_node and not self._is_member_type_wrapped_as_shared_pointer:
+        if not self.is_integral_type_node:
             return "const " + typename + "&"
         return typename
 
     def get_add_methods_declaration(self):
         s = ''
         if self.add_method:
             method = f"""
-                         /**
```

Comment:
Actually I figured out that this file gets generated based on https://github.com/BlueBrain/nmodl/blob/master/src/language/templates/visitors/nmodl_visitor.cpp which might be out of the scope of this PR so feel free to disregard my comment

Reply:
yeah, I was focused on the generated headers useful for Mickael in this PR. But I agree this is a topic that should be addressed at some point.
The generated code based on jinja templates will never be perfectly formatted. ClangFormat is already part of the code generation but only as an option for now
https://github.com/BlueBrain/nmodl/blob/10bf2d21a8402143863aa081a5a6257e8678ba82/src/language/CMakeLists.txt#L10-L15
We could revisit this part of the code so that ClangFormat is always called.

Reply:
I see. I think we can discuss separately the formatting option so from my side this PR is ready to 🚀